master/buildbot/db/logs.py

```diff
@@ -122,17 +152,29 @@ def thd(conn):
             remaining = content.encode('utf-8')
             while remaining:
                 chunk, remaining = self._splitBigChunk(remaining, logid)
-
                 last_line = chunk_first_line + chunk.count('\n')
+
+                # Set the default compressed mode to "raw" id
+                compressed_mode = self.COMPRESSION_MODE["raw"]["id"]
+                # Do we have to compress the chunk?
+                if self.master.config.logCompressionMethod != "raw":
+                    compressed_mode = self.COMPRESSION_MODE[self.master.config.logCompressionMethod]
+                    compressed_chunk = compressed_mode["dumps"](chunk)
+                    # Is it useful to compress the chunk?
+                    if len(chunk) > len(compressed_chunk):
+                        compressed_mode = compressed_mode["id"]
```

Comment:
looks like the else of this check is not tested, as compressed_mode has to be reset to 0